packages/components/src/progress_line/index.jsx

```diff
@@ -0,0 +1,43 @@
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import styles from './index.module.scss';
+
+export const PROGRESS_LINE_WEIGHT_THIN = 'thin';
+export const PROGRESS_LINE_WEIGHT_NORMAL = 'normal';
+
+const ProgressLine = (props) => {
+  const { steps, current, weight, ...cleanProps } = props;
+  const className = clsx(styles.root, props.className, {
+    [styles.thin]: weight === PROGRESS_LINE_WEIGHT_THIN,
+  });
+
+  const chunks = Array.from({ length: steps }, (_, index) => (
+    <li
+      key={index} className={clsx(styles.step, {
+        [styles.active]: index + 1 <= current,
+      })}
+    />
+  ));
+
+  return (
+    <ul {...cleanProps} className={className}>{chunks}</ul>
```

Comment:
Even though it seems like it works, we are just usually going with `.map()` and render function, maybe worth keeping this way?